src/read.c

```diff
@@ -1029,19 +1029,37 @@ display_stinking_cloud_positions(int state)
     }
 }
 
+/* getobj callback for armor object to enchant */
+static int
+armor_enchant_ok(struct obj* obj)
+{
+    if (!obj)
+        return GETOBJ_EXCLUDE;
+
+    if (obj->owornmask & W_ARMOR)
+        return GETOBJ_SUGGEST;
+
+    return GETOBJ_EXCLUDE_INACCESS;
+}
+
 static void
 seffect_enchant_armor(struct obj **sobjp)
 {
     struct obj *sobj = *sobjp;
     register schar s;
     boolean special_armor;
     boolean same_color;
-    struct obj *otmp = some_armor(&g.youmonst);
     boolean sblessed = sobj->blessed;
     boolean scursed = sobj->cursed;
     boolean confused = (Confusion != 0);
     boolean old_erodeproof, new_erodeproof;
 
+    struct obj *otmp = getobj("enchant", armor_enchant_ok, GETOBJ_NOFLAGS);
```

Comment:
Since this makes it a prompting scroll, I would add a "This is an enchant armor scroll." message here before the prompt, similar to charging or genocide scrolls, as well as makeknown(SCR_ENCHANT_ARMOR) for perminvent update prior to the getobj (then possibly remove a makeknown() call later on that previously did the job).